samples/net/wifi/boards/esp32c3_devkitm.conf

```diff
@@ -0,0 +1,18 @@
+CONFIG_WIFI=y
+CONFIG_WIFI_ESP32=y
+CONFIG_HEAP_MEM_POOL_SIZE=98304
+
+CONFIG_NETWORKING=y
+CONFIG_NET_L2_ETHERNET=y
+
+CONFIG_NET_IPV6=n
+CONFIG_NET_IPV4=y
+CONFIG_NET_DHCPV4=y
+
+CONFIG_NET_MGMT=y
+CONFIG_NET_MGMT_EVENT=y
+
+CONFIG_NET_LOG=y
+
+CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4=y
+CONFIG_ESP32_WIFI_STA_RECONNECT=n
```

Comment:
Maybe it should be disabled by default in Kconfig, so that application layer with use of wifi_mgmt API will take care of reconnection?